plugin/pkg/global/resourcereferencemanager/admission.go

```diff
@@ -610,8 +618,8 @@ func (r *ReferenceManager) ensureSecretBindingReferences(ctx context.Context, at
 
 func (r *ReferenceManager) ensureShootReferences(ctx context.Context, attributes admission.Attributes, oldShoot, shoot *core.Shoot) error {
 	if !equality.Semantic.DeepEqual(oldShoot.Spec.CloudProfileName, shoot.Spec.CloudProfileName) {
-		if _, err := r.cloudProfileLister.Get(shoot.Spec.CloudProfileName); err != nil {
-			return err
+		if _, err := utils.GetCloudProfile(r.cloudProfileLister, r.namespacedCloudProfileLister, shoot.Spec.CloudProfileName, shoot.Namespace); err != nil {
```

Comment:
Shouldn't we also consider `shoot.Spec.CloudProfile` here?

Reply:
Yes we should. Please see the other discussion about the `Spec.CloudProfile` field for more information.